tests/utf8.rs

```diff
@@ -279,29 +332,9 @@ fn refuse_invalid_utf8_subcommand_with_allow_external_subcommands() {
     assert_eq!(m.unwrap_err().kind, ErrorKind::InvalidUtf8);
 }
 
-#[test]
-fn allow_invalid_utf8_subcommand_with_allow_external_subcommands() {
```

Comment:
Can you describe a bit on why this is deleted? Just want to make sure we are on the same page.

Reply:
The other tests are about having invalid UTF-8 in the argument list for an external subcommand.   This test case is about the external subcommand itself (ensuring `workspace` in `cargo workspace --help` is UTF-8).

This feature is half-baked.  The validator will allow external subcommands to be invalid UTF-8 but the only API to access the subcommand will panic.

I decided to error instead of continuing the panic.  non-UTF-8 paths are extremely rare.  The likelihood that the external subcommand would have invalid UTF-8 is so low, I think this is safe to keep long term.  We can always see if someone complains and we can add a `_os` API at that time and that would not break compatibility.

Reply:
In that case, please turn this into a `should_panic` test

Reply:
Please explain why we should be allowing this panic to continue and what you are concerned about with my reasons for not continuing it?

Reply:
I think you misunderstood me. I meant, use this test to test the error we are returning at https://github.com/clap-rs/clap/pull/2677/files#diff-cee11e25bbd78f2a4a6dc90846a6e6084923c8ec4d572693290dca3c37797c18R691